apps/maestro/src/features/RegisterRemoteTokens/RegisterRemoteTokens.tsx

```diff
@@ -121,14 +121,49 @@ export const RegisterRemoteTokens: FC<RegisterRemoteTokensProps> = (props) => {
     txState,
   ]);
 
+  const onStellarTxComplete = useCallback(async () => {
+    if (txState.status !== "submitted") return;
+    if (!txState.hash) return;
+
+    const remoteTokens = baseRemoteTokens.map((remoteToken) => ({
+      ...remoteToken,
+      deploymentTxHash: txState.hash,
+    }));
+
+    await recordRemoteTokenDeployment({
+      tokenAddress: props.tokenAddress,
+      chainId: props.originChainId ?? -1,
+      axelarChainId: stellarChainConfig.axelarChainId,
+      deploymentMessageId: txState.hash,
+      remoteTokens,
+    });
+    setTxState({
+      status: "confirmed",
+      hash: txState.hash,
+    });
+  }, [
+    baseRemoteTokens,
+    props.originChainId,
+    props.tokenAddress,
+    recordRemoteTokenDeployment,
+    setTxState,
+    txState,
+  ]);
+
   useEffect(
     () => {
       if (txState.status !== "submitted") return;
-
-      onSuiTxComplete().catch((error) => {
-        logger.error("Failed to record remote token deployment", error);
-        toast.error("Failed to record remote token deployment");
-      });
+      if (props.originChainId === suiChainConfig.id) {
+        onSuiTxComplete().catch((error) => {
+          logger.error("Failed to record remote token deployment", error);
+          toast.error("Failed to record remote token deployment");
+        });
+      } else if (props.originChainId === stellarChainConfig.id) {
+        onStellarTxComplete().catch((error) => {
+          logger.error("Failed to record remote token deployment", error);
+          toast.error("Failed to record remote token deployment");
+        });
+      }
```

Comment:
Create a mapping of chainId and tx complete callback function so it's easier to manage? e.g.

```ts
const txCompleteCallback: Record<string, () => void> = {
     [suiChainConfig.id]: onSuiTxComplete,
     [stellarChainConfig.id]: onStellarTxComplete
}
```

then the usage here should become simpler.